app/views/mirador/_show.html.erb

```diff
@@ -1,5 +1,6 @@
 <% if mirador_displayable?(document) %>
+  <% query_param = @current_query ? "&q=#{@current_query}" : '' %>
```

Comment:
Since this going directly into the URL, if `@current_query` isn't already URL-escaped (and I don't think it is), it should be before it gets written into the link.